### PR TITLE
SELinux Module to allow rootless containers 

### PR DESCRIPTION
**Pull request by KyWa**
Work done by Nvidia and Red Hat can be found below. Ansible has been used to do this and more, but I have added a simple bash script to assist with this portion of it.

https://github.com/NVIDIA/dgx-selinux

### CODE REVIEW

**Comment by KyWa** [comment]
This will allow the `--privileged` flag to be removed from all start-container.sh scripts

File: selinux.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+tmp_dir=$(mktemp -d /tmp/nvidia-selinux-XXXXX)
+
+# Prereqs
+yum install -y git
+
+# Get selinux module source
+git clone https://github.com/NVIDIA/dgx-selinux $tmp_dir/dgx-selinux
+
+# Inject newly compiled SELinux module
+semodule -i $tmp_dir/dgx-selinux/bin/RHEL7/nvidia-container-pp
```
Comment (mjlbach) [inline selinux.sh]: I think this should be nvidia-container.pp instead of nvidia-container-pp

**Comment by mjlbach** [comment]
Thanks for this! Left a single comment. I'm having issues running nvidia-smi though without the --privileged flag still

**Comment by KyWa** [comment]
Forgot to add the core part of this and that is the `restorecon` which sets the contexts.

**Comment by mjlbach** [comment]
@KyWa Still having a few issues:

I get the following error upon the second call to restorecon on F32.

```
❯ sudo restorecon -Rv /dev
/dev/tty2 not reset as customized by admin to unconfined_u:object_r:user_tty_device_t:s0
```

I'm not sure if that's pertinent, but when I try to run a tensorflow docker container, it still can only access the GPU if I'm using sudo and --privileged

```
❯ podman run -it --rm --network=host --security-opt label=type:nvidia_container_t tensorflow/tensorflow:2.1.0-gpu-py3 python -c "import tensorflow as tf; print(tf.reduce_sum(tf.random.normal([1000, 1000])))"

2020-04-28 17:16:19.146361: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libnvinfer.so.6
2020-04-28 17:16:19.147442: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libnvinfer_plugin.so.6
2020-04-28 17:16:19.540543: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcuda.so.1
2020-04-28 17:16:19.541058: E tensorflow/stream_executor/cuda/cuda_driver.cc:351] failed call to cuInit: CUDA_ERROR_NO_DEVICE: no CUDA-capable device is detected
```
```
❯ sudo podman run -it --rm --privileged --network=host --security-opt label=type:nvidia_container_t tensorflow/tensorflow:2.1.0-gpu-py3 python -c "import tensorflow as tf; print(tf.
reduce_sum(tf.random.normal([1000, 1000])))"
2020-04-28 17:17:54.294540: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libnvinfer.so.6
2020-04-28 17:17:54.295641: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libnvinfer_plugin.so.6
2020-04-28 17:17:54.682518: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcuda.so.1
2020-04-28 17:17:54.693568: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one
NUMA node, so returning NUMA node zero
2020-04-28 17:17:54.693849: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1555] Found device 0 with properties:
pciBusID: 0000:01:00.0 name: GeForce GTX 1070 computeCapability: 6.1
```

**Comment by KyWa** [comment]
This is actually a new one for me, but most likely isn't related.
> ```
> ❯ sudo restorecon -Rv /dev
> /dev/tty2 not reset as customized by admin to unconfined_u:object_r:user_tty_device_t:s0
> ```

I will be rebuilding a node later today to test, but were you able to test the initial podman command given and did it behave the same way (not able to use the GPU)?

```
podman run --user 1000:1000 --network=host --security-opt label=type:nvidia_container_t mirrorgooglecontainers/cuda-vector-add:v0.1
```

**Comment by mjlbach** [comment]
@KyWa Yes, I should have posted those results:

**rootless**
```
❯ podman run --user 1000:1000 --network=host --security-opt label=type:nvidia_container_t mirrorgooglecontainers/cuda-vector-add:v0.1
Failed to allocate device vector A (error code CUDA driver version is insufficient for CUDA runtime version)!
[Vector addition of 50000 elements]
```
**root without privileged**
```
❯ sudo podman run --user 1000:1000 --network=host --security-opt label=type:nvidia_container_t mirrorgooglecontainers/cuda-vector-add:v0.1
[sudo] password for michael:
Failed to allocate device vector A (error code no CUDA-capable device is detected)!
[Vector addition of 50000 elements]
```
**privileged without root**
```
❯ podman run --privileged  --user 1000:1000 --network=host --security-opt label=type:nvidia_container_t mirrorgooglecontainers/cuda-vector-add:v0.1
Failed to allocate device vector A (error code CUDA driver version is insufficient for CUDA runtime version)!
[Vector addition of 50000 elements]
```

**root with privileged**
```
❯ sudo podman run --privileged  --user 1000:1000 --network=host --security-opt label=type:nvidia_container_t mirrorgooglecontainers/cuda-vector-add:v0.1
[Vector addition of 50000 elements]
Copy input data from the host memory to the CUDA device
CUDA kernel launch with 196 blocks of 256 threads
Copy output data from the CUDA device to the host memory
Test PASSED
Done
```

**Comment by mjlbach** [comment]
This seems to be working for me now, I cleared my local user podman.conf after the F32 upgrade. Thanks for the contribution :)

**Pull request merged**